cpp/src/arrow/filesystem/s3_test_util.cc

```diff
@@ -105,44 +83,26 @@ Status MinioTestServer::Start() {
 
   ARROW_ASSIGN_OR_RAISE(impl_->temp_dir_, TemporaryDir::Make("s3fs-test-"));
 
-  // Get a copy of the current environment.
-  // (NOTE: using "auto" would return a native_environment that mutates
-  //  the current environment)
-  bp::environment env = boost::this_process::environment();
-  env["MINIO_ACCESS_KEY"] = kMinioAccessKey;
-  env["MINIO_SECRET_KEY"] = kMinioSecretKey;
-  // Disable the embedded console (one less listening address to care about)
-  env["MINIO_BROWSER"] = "off";
-
+  impl_->server_process_ = std::make_unique<util::Process>();
+  ARROW_RETURN_NOT_OK(
+      impl_->server_process_->SetEnv("MINIO_ACCESS_KEY", kMinioAccessKey));
+  ARROW_RETURN_NOT_OK(
+      impl_->server_process_->SetEnv("MINIO_ACCESS_KEY", kMinioAccessKey));
+  ARROW_RETURN_NOT_OK(
+      impl_->server_process_->SetEnv("MINIO_SECRET_KEY", kMinioSecretKey));
+  ARROW_RETURN_NOT_OK(impl_->server_process_->SetEnv("MINIO_BROWSER", "off"));
```

Comment:
Can you keep the comment from above?

Reply:
Ah, I should have not remove the comment. I'll re-add it.